src/passkey_child/passkey_child_assert.c

```diff
@@ -125,6 +156,69 @@ set_assert_options(fido_opt_t up, fido_opt_t uv, fido_assert_t *_assert)
     return ret;
 }
 
+errno_t
+get_assert_auth_data_signature(TALLOC_CTX *mem_ctx, fido_assert_t *assert,
+                               const char **_auth_data,
+                               const char **_signature)
+{
+    const unsigned char *auth_data;
+    const unsigned char *signature;
+    size_t auth_data_len;
+    size_t signature_len;
+    errno_t ret;
+
+    auth_data = fido_assert_authdata_ptr(assert, 0);
+    if (auth_data == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE,
+              "fido_assert_authdata_ptr failed.\n");
+        ret = ENOMEM;
+        goto done;
+    }
+
+    auth_data_len = fido_assert_authdata_len(assert, 0);
+    if (auth_data_len == 0) {
+        DEBUG(SSSDBG_OP_FAILURE,
+              "fido_assert_authdata_len failed.\n");
+        ret = ENOMEM;
+        goto done;
+    }
+
+    *_auth_data = sss_base64_encode(mem_ctx, auth_data, auth_data_len);
```

Comment:
You allocate _auth_data here, but you don't free it in case of error further in this code. You should use tmp_ctx pattern here.

https://sssd.io/contrib/talloc.html#temporary-memory-context

Reply:
Done also for `*_signature` as it follows the same principle.